### ISSUE TITLE
Hangs/Crashes in BBTrack when cloning a track while playing

### ISSUE THREAD

**Issue by zonkmachine** [LMMS/lmms#2693]
Just clone a kicker track when playing and it will hang or crash pretty much all the time.
I don't see this in the stable versions but bisecting hasn't given anything useful. 

```
zonkmachine@zonkmachine:~/builds/lmms/lmms/build$ ./lmms 
...
called Track::getTCO( 0 ), but TCO 0 doesn't exist
QObject: Cannot create children for a parent that is in a different thread.
(Parent is InstrumentTrack(0x7f5af1684a60), parent's thread is QThread(0x267a730), current thread is QThread(0x2b49800)
Segmentation fault (core dumped)
```

```

zonkmachine@zonkmachine:~/builds/lmms/lmms/build$ gdb ./lmms core 
GNU gdb (Ubuntu 7.7.1-0ubuntu5~14.04.2) 7.7.1
...
Core was generated by `./lmms'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0xffffffffffffffe0 in ?? ()
```
`0xffffffffffffffe0` that's `0 - 0x000000000000001f
`

```
(gdb) bt
#0  0xffffffffffffffe0 in ?? ()
#1  0x000000000060dc45 in Pattern::addNote (this=0x7f5af1686de0, _new_note=..., _quant_pos=false) at /home/zonkmachine/builds/lmms/lmms/src/tracks/Pattern.cpp:217
#2  0x000000000060f22f in Pattern::ensureBeatNotes (this=0x7f5af1686de0) at /home/zonkmachine/builds/lmms/lmms/src/tracks/Pattern.cpp:569
#3  0x000000000060d78e in Pattern::init (this=0x7f5af1686de0) at /home/zonkmachine/builds/lmms/lmms/src/tracks/Pattern.cpp:124
#4  0x000000000060d472 in Pattern::Pattern (this=0x7f5af1686de0, _instrument_track=0x7f5af1684a60) at /home/zonkmachine/builds/lmms/lmms/src/tracks/Pattern.cpp:68
#5  0x0000000000605ce9 in InstrumentTrack::createTCO (this=0x7f5af1684a60) at /home/zonkmachine/builds/lmms/lmms/src/tracks/InstrumentTrack.cpp:685
#6  0x0000000000550679 in Track::getTCO (this=0x7f5af1684a60, tcoNum=0) at /home/zonkmachine/builds/lmms/lmms/src/core/Track.cpp:2211
#7  0x00000000004e5326 in BBTrackContainer::lengthOfBB (this=0x28cfc50, _bb=0) at /home/zonkmachine/builds/lmms/lmms/src/core/BBTrackContainer.cpp:107
#8  0x00000000004e512b in BBTrackContainer::play (this=0x28cfc50, _start=..., _frames=205, _offset=51, _tco_num=0) at /home/zonkmachine/builds/lmms/lmms/src/core/BBTrackContainer.cpp:60
#9  0x0000000000601edc in BBTrack::play (this=0x7f5af167b960, _start=..., _frames=205, _offset=51, _tco_num=0) at /home/zonkmachine/builds/lmms/lmms/src/tracks/BBTrack.cpp:463
#10 0x0000000000542387 in Song::processNextBuffer (this=0x295fb60) at /home/zonkmachine/builds/lmms/lmms/src/core/Song.cpp:394
#11 0x000000000051e466 in Mixer::renderNextBuffer (this=0x28d3fc0) at /home/zonkmachine/builds/lmms/lmms/src/core/Mixer.cpp:395
#12 0x000000000051fe2d in Mixer::fifoWriter::run (this=0x2b49800) at /home/zonkmachine/builds/lmms/lmms/src/core/Mixer.cpp:986
#13 0x00007f5afd06b32f in ?? () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#14 0x00007f5afe3e0182 in start_thread (arg=0x7f5abffff700) at pthread_create.c:312
#15 0x00007f5afa9fa47d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111
```

Full backtrace: https://gist.github.com/zonkmachine/62c27a133e817b57e211

**Comment by Fastigium** [comment]
Could this be related to #2470?

**Comment by zonkmachine** [comment]
> Could this be related to #2470?

No. That bug seem to have been introduced in #1704 / 6e68eca6f88a36a7c6fd13d907a8b6fe55069bbc and this issue should be placed in time somewhere around half a year prior to that.

**Comment by zonkmachine** [comment]
I forgot to set milestone for this when I added it. It's still an issue and I think it would be good to see this fixed for 1.2.x . However, it's a bit beyond my capacity as coding goes.
